### PR TITLE
En_hg Decompiled

### PR DESCRIPTION
**Pull request by jpburnett**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Termina's okayest dad (heck even tingle's dad isn't that bad). The supernatural scientist himself is good to go.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Hg/z_en_hg.h [hunk 1]
```diff
@@ -5,9 +5,20 @@
...
-    /* 0x144 */ char unk_144[0x1D4];
+    /* 0x144 */ ColliderCylinder collider;
+    /* 0x190 */ SkelAnime skelAnime;
+    /* 0x1D4 */ EnHgActionFunc actionFunc;
+    /* 0x1D8 */ MtxF unk1D8;
+    /* 0x218 */ s32 unk218;
+    /* 0x21C */ s32 currentAnimation;
+    /* 0x220 */ Vec3s limbDrawTbl[19];
+    /* 0x292 */ Vec3s transitionDrawTbl[19];
+    /* 0x304 */ char pad304[0xC];
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.h:20]:
```suggestion
    /* 0x304 */ UNK_TYPE1 pad304[0xC];
```

**Review by Kenix3 — CHANGES_REQUESTED** [review]
Just one small issue I noticed on a quick review. Good work!

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 2]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+
+void EnHg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHg* this = THIS;
+    s16 phi_s1;
+    s32 i;
+
+    phi_s1 = this->actor.cutscene;
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 36.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06008580, &D_0600260C, this->limbDrawTbl, this->transitionDrawTbl,
+                     0x13);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:132]:
```suggestion
    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06008580, &D_0600260C, this->limbDrawTbl, this->transitionDrawTbl,
                     19);
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 3]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+
+void func_80BCF354(EnHg* this) {
+    func_800BDC5C(&this->skelAnime, animations, 0);
+    this->actionFunc = func_80BCF398;
+}
+
+void func_80BCF398(EnHg* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health == 1) {
+        if ((this->actor.xzDistToPlayer < 200.0f && this->actor.yDistToPlayer < 40.0f) &&
+            (func_800EE29C(globalCtx, 0x1E3U) == 0)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:166]:
```suggestion
            (!func_800EE29C(globalCtx, 0x1E3)) {
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 4]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+}
+
+void func_80BCF4AC(EnHg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/EnHg_Destroy.asm")
+    this->actor.speedXZ = 1.6f;
+    if (!(player->stateFlags2 & 0x08000000)) {
+        if (func_80152498(&globalCtx->msgCtx) == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:187]:
```suggestion
        if (!func_80152498(&globalCtx->msgCtx)) {
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 5]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF4AC.asm")
+void func_80BCF68C(EnHg* this) {
+    func_800BDC5C(&this->skelAnime, animations, 2);
+    this->actionFunc = func_80BCF6D0;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF5F0.asm")
+void func_80BCF6D0(EnHg* this, GlobalContext* globalCtx) {
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:221]:
```suggestion
    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 6]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+        func_80BCF5F0(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF634.asm")
+void func_80BCF710(EnHg* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 0) {
+        if (func_800B84D0(&this->actor, globalCtx)) {
+            func_801518B0(globalCtx, 0x24FU, &this->actor);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:229]:
```suggestion
            func_801518B0(globalCtx, 0x24F, &this->actor);
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 7]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF5F0.asm")
+void func_80BCF6D0(EnHg* this, GlobalContext* globalCtx) {
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount) != 0) {
+        func_80BCF5F0(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF634.asm")
+void func_80BCF710(EnHg* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:227]:
```suggestion
    if (!func_80152498(&globalCtx->msgCtx)) {
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 8]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF7D8.asm")
+void func_80BCF93C(EnHg* this) {
+    this->cutscenes[3] = 0x63;
+    this->cutscenes[2] = 0;
+    this->actionFunc = func_80BCF95C;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF88C.asm")
+void func_80BCF95C(EnHg* this, GlobalContext* globalCtx) {
+    if (func_800EE29C(globalCtx, 0x1E4U) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:283]:
```suggestion
    if (func_800EE29C(globalCtx, 0x1E4)) {
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 9]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+        u32 actionIndex = func_800EE200(globalCtx, 0x1E4U);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:284]:
```suggestion
        u32 actionIndex = func_800EE200(globalCtx, 0x1E4);
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 10]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+                    this->currentAnimation = 1;
+                    func_800BDC5C(&this->skelAnime, animations, 1);
+                    break;
+                case 6:
+                    gSaveContext.weekEventReg[0x4B] |= 0x20;
+                    Actor_MarkForDeath(&this->actor);
+                    break;
+            }
+        } else {
+            if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:320]:
```suggestion
            if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 11]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+    }
+    this->cutscenes[3] = 0x63;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF8A0.asm")
+void func_80BCFC0C(EnHg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF93C.asm")
+    if (this->actor.colChkInfo.health == 1 && !(fabsf(this->actor.yDistToPlayer) >= 80.0f)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:359]: Does `(fabsf(this->actor.yDistToPlayer) < 80.0f))` also match?

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 12]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+                    this->unk218 = 3;
+                } else {
+                    this->unk218 = 1;
+                }
+                func_80BCF88C(this);
+            }
+        } else {
+            if (this->actor.xzDistToPlayer < 60.0f && fabsf(this->actor.yDistToPlayer) < 40.0f) {
+                if ((this->actionFunc != func_80BCF8A0) && (this->actionFunc != func_80BCF95C)) {
+                    if ((gSaveContext.weekEventReg[0x3D] & 2) == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:380]:
```suggestion
                    if (!(gSaveContext.weekEventReg[0x3D] & 2)) {
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 13]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+            if (D_80BD00C8 == 0) {
+                play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            }
+            D_80BD00C8 = 1;
+        } else {
+            D_80BD00C8 = 0;
+        }
+        if (globalCtx->msgCtx.unk1202A == 3) {
+            if (globalCtx->msgCtx.unk1202E == 7 && gSaveContext.playerForm == PLAYER_FORM_HUMAN) {
+                if (INV_CONTENT(0x41) == 0x41) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:370]: Any ideas on what inventory item it's checking for? No worries if you don't

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 14]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+                    this->unk218 = 1;
+                }
+                func_80BCF88C(this);
+            }
+        } else {
+            if (this->actor.xzDistToPlayer < 60.0f && fabsf(this->actor.yDistToPlayer) < 40.0f) {
+                if ((this->actionFunc != func_80BCF8A0) && (this->actionFunc != func_80BCF95C)) {
+                    if ((gSaveContext.weekEventReg[0x3D] & 2) == 0) {
+                        gSaveContext.weekEventReg[0x3D] |= 2;
+                        this->unk218 = NULL;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:382]:
```suggestion
                        this->unk218 = 0;
```
Looks like this is only set to ints, not ptrs.

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 15]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+    this->actionFunc(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80BCF7D8(this, globalCtx);
+    func_80BCFC0C(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 25.0f, 0.0f, 5);
+    func_80BCF778(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/EnHg_Update.asm")
+// override limb
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:406]:
```suggestion
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 16]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+s32 EnHg_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    return 0;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCFE54.asm")
+// post limb
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:411]:
```suggestion
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 17]
```diff
@@ -1,3 +1,9 @@
+/*
+ * File: z_en_hg.c
+ * Overlay: En_Hg
+ * Description: Pamela's Father (Normal)
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:4]: Normal seems a little ambiguous. I'd prefer healed to be a little more explicit.

**Review by hensldm — COMMENTED** [review]
Nice job.

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 18]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF710.asm")
+void func_80BCF88C(EnHg* this) {
+    this->actionFunc = func_80BCF8A0;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF778.asm")
+void func_80BCF8A0(EnHg* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->cutscenes[this->unk218]) != 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:265]:
```suggestion
    if (ActorCutscene_GetCanPlayNext(this->cutscenes[this->unk218])) {
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 19]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit2);
+    if ((gSaveContext.weekEventReg[0x4B] & 0x20) || (gSaveContext.weekEventReg[0x34] & 0x20)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    this->actor.targetMode = 1;
+    this->actor.colChkInfo.health = 0;
+    this->actor.gravity = -1.0f;
+    for (i = 0; i < 4; i++) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:142]:
```suggestion
    for (i = 0; i < ARRAY_COUNT(this->cutscenes); i++) {
```

**Review by AngheloAlf — COMMENTED** [review]
Good job!
I left a few small suggestions

File: linker_scripts/object_script.txt [hunk 20]
```diff
@@ -722,3 +722,15 @@ D_06000968 = 0x06000968;
...
+D_06008580 = 0x06008580;
+D_0600260C = 0x0600260C;
+D_06005E28 = 0x06005E28;
+D_06009D44 = 0x06009D44;
+D_0600A164 = 0x0600A164;
+D_06000370 = 0x06000370;
+D_06001138 = 0x06001138;
+D_060015D4 = 0x060015D4;
+D_06001960 = 0x06001960;
+D_0600AE1C = 0x0600AE1C;
```
Comment (engineer124) [inline linker_scripts/object_script.txt:736]: Could you move this block up so its in proper alphabetical order?

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 21]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+static CollisionCheckInfoInit2 sColChkInfoInit2 = {
+    0, 0, 0, 0, 0x80,
+};
+
+static ActorAnimationEntry animations[] = {
+    { &D_0600260C, 1.0f, 0.0f, 0.0f, 0x00, -4.0f }, { &D_06009D44, 1.0f, 0.0f, 0.0f, 0x00, -4.0f },
+    { &D_0600A164, 1.0f, 0.0f, 0.0f, 0x00, -4.0f }, { &D_06000370, 1.0f, 0.0f, 0.0f, 0x02, 0.0f },
+    { &D_06001138, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },  { &D_060015D4, 1.0f, 0.0f, 0.0f, 0x02, 0.0f },
+    { &D_06001960, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },  { &D_0600AE1C, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },
+};
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:120]:
```suggestion
static ActorAnimationEntry animations[] = {
    { &D_0600260C, 1.0f, 0.0f, 0.0f, 0, -4.0f }, { &D_06009D44, 1.0f, 0.0f, 0.0f, 0, -4.0f },
    { &D_0600A164, 1.0f, 0.0f, 0.0f, 0, -4.0f }, { &D_06000370, 1.0f, 0.0f, 0.0f, 2, 0.0f },
    { &D_06001138, 1.0f, 0.0f, 0.0f, 0, 0.0f },  { &D_060015D4, 1.0f, 0.0f, 0.0f, 2, 0.0f },
    { &D_06001960, 1.0f, 0.0f, 0.0f, 0, 0.0f },  { &D_0600AE1C, 1.0f, 0.0f, 0.0f, 0, 0.0f },
};
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 22]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+
+u32 D_80BD00C8 = 0;
+
+void EnHg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHg* this = THIS;
+    s16 phi_s1;
+    s32 i;
+
+    phi_s1 = this->actor.cutscene;
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 36.0f);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:130]:
```suggestion
    EnHg* this = THIS;
    s16 phi_s1 = this->actor.cutscene;
    s32 i;

    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 36.0f);
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 23]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+
+void func_80BCF398(EnHg* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health == 1) {
+        if ((this->actor.xzDistToPlayer < 200.0f && this->actor.yDistToPlayer < 40.0f) &&
+            (func_800EE29C(globalCtx, 0x1E3U) == 0)) {
+            func_80BCF468(this);
+        }
+        if ((gSaveContext.sceneSetupIndex == 0 && globalCtx->csCtx.unk_12 == 0) &&
+            (globalCtx->csCtx.frames == 0x14 || globalCtx->csCtx.frames == 0x3C)) {
+            Audio_PlayActorSound2(&this->actor, 0x3AB8);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:171]:
```suggestion
            Audio_PlayActorSound2(&this->actor, NA_SE_EN_HALF_REDEAD_SURPISE);
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c (hunk 4 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Hg/z_en_hg.c:187]: This if could probably be merged with the one above with an `&&`

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 24]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+                case 3:
+                    this->cutscenes[2] = 0;
+                    this->currentAnimation = 5;
+                    func_800BDC5C(&this->skelAnime, animations, 5);
+                    break;
+                case 4:
+                    this->cutscenes[2] = 0;
+                    this->currentAnimation = 7;
+                    if ((this->unk218 == 1) || (this->unk218 == 3)) {
+                        func_8019F128(0x3ABA);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:306]:
```suggestion
                        func_8019F128(NA_SE_EN_HALF_REDEAD_TRANS);
```
`func_8019F128` seems to be from an audio file, and the sfx matches up

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 25]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF68C.asm")
+void func_80BCF778(EnHg* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health == 1) {
+        if (this->actionFunc == func_80BCF4AC || this->actionFunc == func_80BCF6D0 ||
+            this->actionFunc == func_80BCF634) {
+            func_800B9010(&this->actor, 0x32B7);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:240]:
```suggestion
            func_800B9010(&this->actor, NA_SE_EN_HALF_REDEAD_LOOP - SFX_FLAG);
```
also an audio function

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 26]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+            case 4:
+                func_800B9010(&this->actor, 0x32B7);
+                break;
+            case 5:
+            case 6:
+                func_800B9010(&this->actor, 0x32B9);
+                break;
+            case 7:
+                if ((this->unk218 == 0) || (this->unk218 == 2)) {
+                    func_800B9010(&this->actor, 0x32B9);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:344]:
```suggestion
                func_800B9010(&this->actor, NA_SE_EN_HALF_REDEAD_LOOP - SFX_FLAG);
                break;
            case 5:
            case 6:
                func_800B9010(&this->actor, NA_SE_EN_HALF_REDEAD_SCREAME - SFX_FLAG);
                break;
            case 7:
                if ((this->unk218 == 0) || (this->unk218 == 2)) {
                    func_800B9010(&this->actor, NA_SE_EN_HALF_REDEAD_SCREAME - SFX_FLAG);
```

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 27]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF93C.asm")
+    if (this->actor.colChkInfo.health == 1 && !(fabsf(this->actor.yDistToPlayer) >= 80.0f)) {
+        if (player->stateFlags2 & 0x08000000) {
+            if (D_80BD00C8 == 0) {
+                play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            }
+            D_80BD00C8 = 1;
+        } else {
+            D_80BD00C8 = 0;
+        }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:367]:
```suggestion
            if (!D_80BD00C8) {
                play_sound(NA_SE_SY_TRE_BOX_APPEAR);
            }
            D_80BD00C8 = true;
        } else {
            D_80BD00C8 = false;
        }
```
This looks bool

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c [hunk 28]
```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
...
+};
+
+static ActorAnimationEntry animations[] = {
+    { &D_0600260C, 1.0f, 0.0f, 0.0f, 0x00, -4.0f }, { &D_06009D44, 1.0f, 0.0f, 0.0f, 0x00, -4.0f },
+    { &D_0600A164, 1.0f, 0.0f, 0.0f, 0x00, -4.0f }, { &D_06000370, 1.0f, 0.0f, 0.0f, 0x02, 0.0f },
+    { &D_06001138, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },  { &D_060015D4, 1.0f, 0.0f, 0.0f, 0x02, 0.0f },
+    { &D_06001960, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },  { &D_0600AE1C, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },
+};
+
+u32 D_80BD00C8 = 0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hg/z_en_hg.c:122]:
```suggestion
static u32 D_80BD00C8 = false;
```

**Review by engineer124 — COMMENTED** [review]
Great work! I've left my feedback

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c (hunk 11 above)
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Hg/z_en_hg.c:359]: It does not :/

File: src/overlays/actors/ovl_En_Hg/z_en_hg.c (hunk 17 above)
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Hg/z_en_hg.c:4]: That is fair, I updated it to be Human if that makes it a little clearer

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**